### PR TITLE
Improve .gitignore

### PR DESCRIPTION
**Pull request by vasslitvinov**
With the goal of keeping the output of `git status` clean
after running a paratest on the tree, adjust .gitignore:

* Update the paths to ignored files to match #16110 and #17070
* Add a directory "hi", also to .cleanfiles, to match #15920
* Add *.bc for llvm-enabled runs.

Going forward, we should populate .gitignore
in individual directories, rather than adding to the single
.gitignore in the root directory.

### CODE REVIEW

**Comment by vasslitvinov** [comment]
@bradcray does this addition of `chplvis/hi` make sense to you?
@mppf did I get the *.bc situation right?

File: .gitignore [hunk 1]
```diff
@@ -157,6 +158,7 @@ tags
 /test/compflags/thomasvandoren/chpldoc/*.doc.good
 /test/compflags/thomasvandoren/chpldoc/docs/
 
+/test/compflags/waynew/*.bc
```
Comment (mppf) [inline .gitignore:161]: this change seems good to me

**Comment by bradcray** [comment]
I generally prefer putting `.gitignore` files into the directories that generate the files rather than the top-level `.gitignore` file so that if/when the directories are moved around, the `.gitignore` files will move with them and continue to work.  I'm not sure who started the practice of using this global file, but I'd like to invent a time machine so that I could go back and mess with the space-time continuum so that we didn't ever start this approach.

**Comment by bradcray** [comment]
With respect to the specific question about `chplvis/hi`, yes, that sounds good, thanks.

**Comment by mppf** [comment]
> I'd like to invent a time machine so that I could go back and mess with the space-time continuum so that we didn't ever start this approach.

I don't think it's *that* hard to push all of the .gitignore entries into subdirectory .gitignore files

**Comment by bradcray** [comment]
Oh, I completely agree.  I just feel like I'm saying all the time "let's please stop doing this," but we keep on doing it.

**Pull request merged**